patches/minecraft/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -600,6 +600,14 @@
           if (this.f_20935_.m_41781_()) {
              this.m_21329_();
           }
+@@ -2996,6 +_,7 @@
+    }
+ 
+    public void m_5810_() {
++      if (this.m_6117_() && !this.f_20935_.m_41619_()) this.f_20935_.onStopUsing(this, f_20936_);
+       if (!this.f_19853_.f_46443_) {
```

Comment:
I think this is missing the use case in Player.hurtCurrentlyUsedShield
The documentation is also a little unlcear, it should state that it is called in ADDITION to those events.
To make it clear that both will be fired.
Also, would this be a good time to add a start using item level callback to mirror this?

Reply:
I believe Player.hurtCurrentlyUsedShield just calls this method directly, hence me putting the patch in LivingEntity. I'll double check that when I get a chance.

I cannot think of any usecase for a start using item callback at the item level, since you start using an item by calling a method in your own item code. Typically that is just in a single method as well, `Item#use`, if you need to do it in multiple places you can always add your own function to call in the mod instead of needing a forge change.

Doing it at the event level might be reasonable (with an event for stop using as well), but I feel that would be better left to a future PR if desired as the usecases are quite different (and personally, I cannot think of a good usecase for detecting when another mod starts using their item)

Reply:
Had a chance to look at the code, I believe `hurtCurrentlyUsedShield` might be the same sort of case as [MC-231097](https://bugs.mojang.com/browse/MC-231097) where you actually never stop using the item when its broken. I could easily add a patch into that method to call `stopUsingItem` when the shield breaks to fix that though, hard to say whether that is in scope as its another vanilla bug to fix (arguably belongs in #9344)